File: Inventario.cs
using System;
using System.Collections.Generic;


public class Inventario
{

public List< Producto> Listadeproductos {get; set; }
}
public Inventario ()
{
ListadeProductos = new List<Producto> ();
Producto a = new Product ("001" , "iPhoneX" , 0);
Producto b = new Product ("002" , "Laptop Dell" , 5);
Producto c = new Product ("003" , "Monitor samsung" , 2);
Producto d = new Product ("004" , "Mouse" , 100);
Producto e = new Product ("005" , "Headset" , 25);

ListadeProductos.Add(a)
ListadeProductos.Add(b)
ListadeProductos.Add(c)
ListadeProductos.Add(d)
ListadeProductos.Add(e)
}
private void listarProductos(){
    console.Clear ();
    console.WriteLine("");
     console.WriteLine("Listado de Productos");
     console.WriteLine("********************");
     console.WriteLine("Codigo, Descripción y Existencia");

    
    foreach (var product in ListadeProductos)
    {
       console.WriteLine(productos.codigo + " | " + producto.Descripcion + " | " + producto.Existencia.ToString()); 
    }
    Console.ReadLine();
}

private void movimientoInventario(string codigo, int cantidad, string tipoMovimiento){
   foreach (var producto in ListadodeProductos)
    {
    if (producto.Codigo == codigo){
            if (tipoMovimiento == "+"){
                productos.Existencia = producto.Existencia + cantidad;
            } else {
                productos.Existencia = producto.Existencia - cantidad
                ;
            }
        }  
    }
}

public void ingresoDeInventario() {
    string codigo = "";
   string cantidad = "";

    console.Clear ();
    console.WriteLine("");

     console.WriteLine("Ingreso de Productos al Inventario");
     console.WriteLine("**********************************");
  console.WriteLine("Ingrese el codigo del producto");
  codigo = Console.ReadLine();
      console.WriteLine("Ingrese la cantidad del producto");
       cantidad = Console.ReadLine();
     
     movimientoInventario(codigo, Int32.Parse(cantidad), "+");
    }
}
private void listarProductos(){
    console.Clear ();
    console.WriteLine("");
     console.WriteLine("Listado de Productos");
     console.WriteLine("********************");
     console.WriteLine("Codigo, Descripción y Existencia");
}
   foreach (var producto in ListadeProductos)
        {
            Console.WriteLine(producto.Codigo + " | " + producto.Descripcion + " | " + producto.Existencia.ToString());
        }

        Console.ReadLine();
    }

    private void movimientoInventario(string codigo, int cantidad, string tipoMovimiento) {
        foreach (var producto in ListadeProductos)
        {
            if (producto.Codigo == codigo) {
                if (tipoMovimiento == "+") {
                    producto.Existencia = producto.Existencia + cantidad;
                } else {
                    producto.Existencia = producto.Existencia - cantidad;
                }
            }         
        }
    }

    public void ingresoDeInventario() {
        string codigo = "";
        string cantidad = "";

        Console.Clear();
        Console.WriteLine();

        Console.WriteLine("Ingreso de Productos al Inventario");
        Console.WriteLine("**********************************");
        Console.Write("Ingrese el codigo del producto: ");
        codigo = Console.ReadLine();
        Console.Write("Ingrese la cantidad del producto: ");
        cantidad = Console.ReadLine();

        movimientoInventario(codigo, Int32.Parse(cantidad), "+");
    }
} 

    

File: Inventario.cpp
#include <iostream>
#include <string>

using namespace std;

string productos[5][3] = {
{ "001", "iPhoneX", "0" },
       { "002", "Laptop Dell", "5" },
       { "003", "Monitor Samsung", "2" },
{ "004", "Mouse", "100" },
  { "004", "Mouse", "100" },
};

void listarProductos(){
    system ("cls");
    cout << endl;
    cout << "***************ZavMur***************" << endl;
    cout << "*       Listado de Productos       *" << endl;
    cout << "***************ZavMur***************" << endl;
    cout << "***************ZavMur***************" << endl;
    cout << "* Codigo, Descripcion y existencia *" << endl;
    cout << "***************ZavMur***************" << endl;
    cout << endl;

    for (int i = 0; i < 5; i++)
    {
        cout << productos[i][0] << " | " << productos[i][1]<< " | " << productos[i][2] << endl;
    }  
}

void movimientoInventario(string codigo, int cantidad, string tipoMovimiento){
    for (int i = 0; i < 5; i++)
    {
        if (productos[i, 0] == codigo) {
            if (tipoMovimiento == "+"){
               productos[i, 2] = (Int32.Parse(productos[i, 2]) + cantidad).ToString();
            } else {
                productos[i, 2] = (Int32.Parse(productos[i, 2]) - cantidad).ToString();
            }
        }
    }
}
void ingresoDeInventario(){
    string codigo = "";
    int cantidad = 0;

    system ("cls");
    cout << endl;
    cout << "****************ZavMur****************" << endl;
    cout << "* Ingreso de Productos al Inventario *" << endl;
    cout << "****************ZavMur****************" << endl;
    cout << "Ingrese el codigo del producto: ";
    cin >> codigo;
    cout << endl;
    cout << "Ingrese la cantidad del producto:  ";
    cin >> cantidad;
    cout << endl;

    movimientoInventario (codigo, cantidad, "+");
}
void salidaDeInventario(){
    string codigo = "";
    int cantidad = 0;

    system ("cls");
    cout << endl;
    cout << "****************ZavMur***************" << endl;
    cout << "* Salida de Productos al Inventario *" << endl;
    cout << "****************ZavMur***************" << endl;
    cout << "Ingrese el codigo del producto: ";
    cin >> codigo;
    cout << endl;
    cout << "Ingrese la cantidad del producto:  ";
    cin >> cantidad;
    cout << endl;

    movimientoInventario (codigo, cantidad, "-");
}
void transferenciaDeProductoRecibido(){
    string codigo = "";
    int cantidad = 0;

    system ("cls");
    cout << endl;
    cout << "*****************ZavMur*****************" << endl;
    cout << "* Transferencia De Productos Recibidos *" << endl;
    cout << "*****************ZavMur*****************" << endl;
    cout << "Ingrese el codigo del producto: ";
    cin >> codigo;
    cout << endl;
    cout << "Ingrese la cantidad del producto:  ";
    cin >> cantidad;
    cout << endl;

    movimientoInventario (codigo, cantidad, "+");
}
void trasferenciaDeProductosEnviados(){
    string codigo = "";
    int cantidad = 0;

    system ("cls");
    cout << endl;
    cout << "****************ZavMur****************" << endl;
    cout << "* Trasferencia De Productos Enviados *" << endl;
    cout << "****************ZavMur****************" << endl;
    cout << "Ingrese el codigo del producto: ";
    cin >> codigo;
    cout << endl;
    cout << "Ingrese la cantidad del producto:  ";
    cin >> cantidad;
    cout << endl;

    movimientoInventario (codigo, cantidad, "-");
}

int main(int argc, char const *argv[])
{

    int opcion = 0;

    while (true) {
        system ("cls");

        cout << endl;
        cout << "*********ZavMur*********" << endl;
        cout << "* Programa de Inventarios *" << endl;
        cout << "*********ZavMur*********" << endl;
        cout << endl;
        cout << "1 - Productos" << endl;
        cout << "2 - Entrada de Inventario" << endl;
        cout << "3 - Salida de Inventario" << endl;
        cout << "4 - Transferencia de Productos Recibidos de la Tienda: " << endl;
        cout << "5 - Transferencia de Productos Enviados a la Tienda:" << endl;
        cout << "0 - Salir" << endl;
        cout << "Ingrese una opcion del menu: ";
        cin >> opcion;

        switch (opcion)
        {
        case 1: 
            listarProductos();            
            break;
        case 2:
            ingresoDeInventario();
            break;
        case 3:
            salidaDeInventario();
            break;
        case 4:
            transferenciaDeProductoRecibido();
            break;
        case 5:
            trasferenciaDeProductosEnviados();
            break;        
        default:
        break;
        }

        system ("pause");
        cout << endl;
        if (opcion == 0)
        {
            break;
        }
        

        }
    return 0;
}
